pkg/syncer/shared/helpers.go

```diff
@@ -71,3 +71,12 @@ func GetDNSID(clusterName logicalcluster.Name, syncTargetUID types.UID, syncTarg
 
 	return fmt.Sprintf("kcp-dns-%s-%s-%s", syncTargetName, uid36hash[:8], workspace36hash[:8])
 }
+
+func Contains[T comparable](elems []T, v T) bool {
```

Comment:
FWIW these kinds of methods are more than likely to make the code significantly less performant [xref](https://planetscale.com/blog/generics-can-make-your-go-code-slower) - the non-generic `for` loop might just be a sensible thing to keep around.

Reply:
Fixed. In fact this method was not used anymore. I dropped it.

Reply:
Thanks for the xref BTW